src/lambda/expression.rs

```diff
@@ -36,43 +39,41 @@ pub enum Context {
   Path(Vec<String>),
 }
 
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 pub enum Unsafe {
-  Http(
-    RequestTemplate,
-    Option<GroupBy>,
-    Option<Arc<DataLoader<HttpDataLoader, NoCache>>>,
-  ),
+  Http {
+    req_template: RequestTemplate,
+    group_by: Option<GroupBy>,
+    dl_id: Option<DataLoaderId<HttpDataLoader>>,
+  },
   GraphQLEndpoint {
     req_template: GraphqlRequestTemplate,
     field_name: String,
     batch: bool,
-    data_loader: Option<Arc<DataLoader<GraphqlDataLoader, NoCache>>>,
+    dl_id: Option<DataLoaderId<GraphqlDataLoader>>,
   },
   JS(Box<Expression>, String),
 }
 
-impl Debug for Unsafe {
+pub struct DataLoaderId<T>(pub usize, PhantomData<T>);
+
+impl<T> DataLoaderId<T> {
+  pub fn new(index: usize) -> Self {
+    Self(index, PhantomData)
+  }
+}
+
+impl<T> Clone for DataLoaderId<T> {
+  fn clone(&self) -> Self {
+    *self
+  }
+}
+
+impl<T> Copy for DataLoaderId<T> {}
```

Comment:
does using PhantomData somehow affects `#[derive]` ?

Reply:
yes, `#[derive]` expects the type parameter T to also implement the trait that we want to derive

Reply:
type safety in such cases helps to prevent wrong usages and helps with code understanding. While this case is pretty simple to understand, it still requires more mental energy to remember what usize is all about and how to use it - i.e. acquire it only from that function, pass it only to other. While new_type put heavy constraints on what could be done and prevents possible misuse in future, especially for devs who didn't work with that code before. 

And it's totally possible to parameterize here as we're using different dataloaders types in different expressions and therefore always know the type on compilation